sdk/appconfiguration/app-configuration/test/public/index.spec.ts

```diff
@@ -1140,4 +1155,304 @@ describe("AppConfigurationClient", () => {
+      if (isSecretReference(actual)) {
+        assert.equal(
+          actual.key,
+          expected.key,
+          "Key from the response from get request is not as expected"
+        );
+        assert.equal(actual.secretId, expected.secretId);
+        assert.equal(actual.isReadOnly, expected.isReadOnly);
+        assert.equal(actual.label, expected.label);
+        assert.equal(actual.contentType, expected.contentType);
+      }
+    }
+
+    let addResponse: AddConfigurationSettingResponse;
+    let baseSetting: SecretReference;
+    beforeEach(async () => {
+      baseSetting = getBaseSetting();
+      addResponse = await client.addConfigurationSetting(baseSetting);
+    });
+
+    afterEach(async () => {
+      await client.deleteConfigurationSetting({
+        key: baseSetting.key
+      });
+    });
+
+    it("can add and get SecretReference", async () => {
+      assertSecretReferenceProps(addResponse, baseSetting);
+      const getResponse = await client.getConfigurationSetting({
+        key: baseSetting.key,
+        label: baseSetting.label
+      });
+      assertSecretReferenceProps(getResponse, baseSetting);
+    });
+
+    it("can add and update SecretReference", async () => {
+      const getResponse = await client.getConfigurationSetting({
+        key: baseSetting.key,
+        label: baseSetting.label
+      });
+      const newSecretId = `https://vault_name.vault.azure.net/secrets/${generateUuid()}`;
+
+      assertSecretReferenceProps(getResponse, baseSetting);
+      if (isSecretReference(getResponse)) {
+        getResponse.secretId = newSecretId;
+      }
+
+      const setResponse = await client.setConfigurationSetting(getResponse);
+      assertSecretReferenceProps(setResponse, {
+        ...baseSetting,
+        secretId: newSecretId
+      });
+
+      const getResponseAfterUpdate = await client.getConfigurationSetting({
+        key: baseSetting.key,
+        label: baseSetting.label
+      });
+      assertSecretReferenceProps(getResponseAfterUpdate, {
+        ...baseSetting,
+        secretId: newSecretId
+      });
+    });
+
+    it("can add, list and update multiple SecretReferences", async () => {
+      const secondSetting = {
+        ...baseSetting,
+        key: `${baseSetting.key}-2`
+      };
+      const newSecretId = `https://vault_name.vault.azure.net/secrets/${generateUuid()}`;
+      await client.addConfigurationSetting(secondSetting);
+
+      let numberOFSecretReferencesReceived = 0;
+      for await (const setting of client.listConfigurationSettings({
+        keyFilter: `${baseSetting.key}*`
+      })) {
+        numberOFSecretReferencesReceived++;
+        if (setting.key === baseSetting.key) {
+          assertSecretReferenceProps(setting, baseSetting);
+          await client.setConfigurationSetting({
+            ...baseSetting,
+            secretId: newSecretId
+          } as SecretReference);
```

Comment:
I don't like that I need to cast it here.
Add overloads for add and set methods to accept FeatureFlags and SecretReferences?

@richardpark-msft

Reply:
Adding the types seems like it'd be good. I wonder if the intellisense will still be intelligible or will it just suggest "all" fields from all the different kinds of settings?

Reply:
Hmm.. I was trying to play with the overloads.. (dejavu.. from service-bus receiveMode and receivers)

I couldn't get the IntelliSense to work the way I expected to. I'll leave it like this, for now, and will investigate more for the next preview.